### PR TITLE
Add nsmenu title

### PR DESCRIPTION
**Pull request by terhechte**
This is an initial fix for #1835.

It uses `::alloc().initWithTitle(...)` to initialize `NSMenu` items with a correct title.
However, it is not clear to me yet what the behaviour would be for the `Menu` that is exposed from the `druid` crate.
Right now I'm calling `Menu("")` but I suppose that's wrong. My guess would be that the public druid API would also expose a title property? But then it would affect all platforms. Open to suggestions here :)

### CODE REVIEW

**Comment by cmyr** [comment]
I think that initing with an empty string and then modifying it afterwards to the correct title should be okay?

**Comment by terhechte** [comment]
Updated it accordingly to use an empty string and added `set_title` which only works on macOS and is a NOP on the other platforms

File: druid-shell/examples/shello.rs [hunk 1]
```diff
@@ -123,7 +123,8 @@ impl WinHandler for HelloState {
 
 fn main() {
     tracing_subscriber::fmt().init();
-    let mut file_menu = Menu::new("File Menu");
+    let mut file_menu = Menu::new();
+    file_menu.set_title("File Menu");
```
Comment (kud1ing) [inline druid-shell/examples/shello.rs:127]: maybe just "File"?

**Comment by cmyr** [comment]
Okay so I'm looking at this again, and I want to double-check my understanding:

- druid-shell's `Menu::new` method is only called for root menus; that means the root application menu, as well as root popup menus.
- all other items (such as the file menu) are created with the `add_item` and `add_dropdown`, which do correctly set titles

And in both the case of. the root menu and of a context menu, the title is (I believe) unused?

Ultimately I'm wondering if we need to bother with the whole `set_title` addition. My main concern with #1835 isn't with the title not being set, it's with a missing `init` call that may be performing other important work. If we can just init with an empty string and leave it at that, I'd be happy for the simplification..

**Comment by terhechte** [comment]
@cmyr I agree. I've removed the `set_title` implementations. Just using an empty string should be enough.

**Review by cmyr — APPROVED** [review]
Thanks!

**Pull request merged**